bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/HttpCacheConfigImpl.java

```diff
@@ -0,0 +1,131 @@
+package com.adobe.acs.commons.httpcache.config.impl;
+
+import com.adobe.acs.commons.httpcache.config.HttpCacheConfig;
+import com.adobe.acs.commons.httpcache.store.HttpCacheStore;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.*;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Config class for http cache. Modelled as OSGi config factory. Every factory instance is tied to a request URI.
+ * Parameters such as selection of cache store, user groups, invalidation details shall also be configured.
+ */
+@Component(label = "ACS AEM Commons - HTTP Cache - Cache config",
+           description = "Config for request URI pattern that has to be cached. Each config is tied to a single " +
+                   "request URI pattern.",
+           configurationFactory = true,
+           metatype = true)
+@Service
+public class HttpCacheConfigImpl implements HttpCacheConfig {
+    private static final Logger log = LoggerFactory.getLogger(HttpCacheConfigImpl.class);
+
+    @Property(label = "Request URI pattern",
```

Comment:
2 reasons for this multiple configs.
(1) Ability to state where this cached response for the request should be persisted. For example, I could say all zip (.*).zip to be persisted in disk and all Json  (.*).json to be in-mem at the same time.
(2) Ability to invalidate the cache selectively. Example - I could put up a config saying all request for product data /content/mysite(.*).product-data.json to be invalidated for any changes in /etc/my-site/my-product-data without disturbing other cached items.